README.md

```diff
@@ -23,396 +23,373 @@ Stable release versions are manually released from the main branch.
-- `unbondingScript`: The script that allows on-demand unbonding.
-  Requires the staker's signature and the covenant committee's signatures.
-- `slashingScript`: The script that enables slashing.
-  It requires the staker's signature and in this phase the staker should not sign it.
-- `dataEmbedScript`: An `OP_RETURN` script containing all required data to
-  identify and verify the transaction as a staking transaction.
+#### Providers
 
-### Create a staking transaction
+The library requires two provider implementations to handle transaction signing:
 
-Using the Bitcoin staking scripts, you can generate a Bitcoin staking
-transaction and later sign it using a supported wallet's method.
-In this instance, we use the `btcWallet.signTransaction()` method.
+1. **Bitcoin Provider**: Handles Bitcoin transaction and message signing
+2. **Babylon Provider**: Handles Babylon transaction signing
 
 ```ts
-import { stakingTransaction } from "@babylonlabs-io/btc-staking-ts";
-import { Psbt, Transaction } from "bitcoinjs-lib";
-
-// stakingTransaction constructs an unsigned BTC Staking transaction
-const unsignedStakingPsbt: {psbt: Psbt, fee: number} = stakingTransaction(
-  scripts: {
-    timelockScript,
-    unbondingScript,
-    slashingScript,
-    dataEmbedScript
-  },
-  stakingAmount,
-  changeAddress,
-  inputUTXOs,
-  network(),
-  feeRate,
-  btcWallet.isTaproot ? btcWallet.publicKeyNoCoord() : undefined,
-  lockHeight,
-);
-
-const signedStakingPsbt = await btcWallet.signPsbt(unsignedStakingPsbt.psbt.toHex());
-const stakingTx = Psbt.fromHex(signedStakingPsbt).extractTransaction();
+export interface BtcProvider {
+  // Sign a PSBT
+  signPsbt(signingStep: SigningStep, psbtHex: string): Promise<string>;
+  // Sign a message using the ECDSA type
+  // This is optional and only required if you would like to use the 
+  // `createProofOfPossession` function
+  signMessage?: (
+    signingStep: SigningStep, message: string, type: "ecdsa" || "bip322-simple"
+  ) => Promise<string>;
+}
+
+export interface BabylonProvider {
+  signTransaction: <T extends object>(
+    signingStep: SigningStep,
+    msg: {
+      typeUrl: string;
+      value: T;
+    }
+  ) => Promise<Uint8Array>
+}
 ```
 
-Public key is needed only if the wallet is in Taproot mode, for `tapInternalKey`.
+### Post-Staking Registration
 
-### Create unbonding transaction
+This refers to registration of an already existing BTC staking transaction into 
+Babylon chain network. 
+For more details, refer to the [Babylon node documentation](https://github.com/babylonlabs-io/babylon/blob/release/v1.x/docs/register-bitcoin-stake.md#21-post-staking-registration).
 
-The staking script allows users to on-demand unbond their locked stake before
-the staking transaction timelock expires, subject to an unbonding period.
-
-The unbonding transaction can be created as follows:
+> **Note**: This method focuses on creating Babylon chain transactions, 
```

Comment:
Which method? The post-staking registration protocol, the code method that is code included below?